app/services/processors/mappings.rb

```diff
@@ -35,9 +35,13 @@ def self.create(specification, user)
     def self.create_terms(mapping, terms)
       terms.each do |term|
         term = Term.find(term[:id])
+        custom_uri = "desm:#{term.uri}"
+
+        # Do not create this term if there's already one with the same uri
+        next if MappingTerm.find_by(uri: custom_uri)
```

Comment:
Are the mappings immutable? If not, how about upserting them via `find_or_initialize_by`?

Reply:
@excelsior by reading the [official documentation](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by) of it, I would think it twice. What do you think?

```
Please note this method is not atomic, it runs first a SELECT, and if there are no results an INSERT is attempted. If there are other threads or processes there is a race condition between both calls and it could be the case that you end up with two similar records.
```